src/libsyncengine/jobs/network/uploadjob.cpp

```diff
@@ -146,41 +146,41 @@ void UploadJob::setQueryParameters(Poco::URI &uri, bool &canceled) {
     canceled = false;
 }
 
+namespace details {
+inline bool isLink(LinkType linkType) {
+    return linkType == LinkTypeSymlink || linkType == LinkTypeHardlink ||
+           (linkType == LinkTypeFinderAlias && OldUtility::isMac()) || (linkType == LinkTypeJunction && OldUtility::isWindows());
+}
+}  // namespace details
+
 void UploadJob::setData(bool &canceled) {
+    canceled = true;
+
     ItemType itemType;
     if (!IoHelper::getItemType(_filePath, itemType)) {
-        LOGW_WARN(KDC::Log::instance()->getLogger(), L"Error in IoHelper::getItemType - path=" << Path2WStr(_filePath).c_str());
-        canceled = true;
+        LOGW_WARN(_logger, L"Error in IoHelper::getItemType - " << Utility::formatSyncPath(_filePath).c_str());
         return;
     }
 
     if (itemType.ioError == IoErrorNoSuchFileOrDirectory) {
-        LOGW_DEBUG(KDC::Log::instance()->getLogger(), L"Item does not exist anymore - path=" << Path2WStr(_filePath).c_str());
-        canceled = true;
+        LOGW_DEBUG(_logger, L"Item does not exist anymore - " << Utility::formatSyncPath(_filePath).c_str());
         return;
     }
 
     if (itemType.ioError == IoErrorAccessDenied) {
-        LOGW_DEBUG(KDC::Log::instance()->getLogger(), L"Item misses search permission - path=" << Path2WStr(_filePath).c_str());
+        LOGW_DEBUG(_logger, L"Item misses search permission - " << Utility::formatSyncPath(_filePath).c_str());
         canceled = true;
```

Comment:
Canceled is already true with your latest changes. For readability sake, we might want to remove this line

Reply:
Good catch, thanks! Done.